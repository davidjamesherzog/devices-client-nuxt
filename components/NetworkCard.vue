<template>
  <div class="column card-width">
    <div class="card">
      <header class="card-header">
        <p class="card-header-title has-text-grey" data-card-title>
          {{ network.name }}
        </p>
      </header>
      <div class="card-content">
        <div class="content has-text-centered">
          <b-icon
            :icon="type"
            size="is-large"
            type="is-primary"
            data-card-icon
          />
        </div>
      </div>
      <footer class="card-footer">
        <div class="card-footer-item">
          <span data-card-description>
            {{ network.description }}
          </span>
        </div>
      </footer>
    </div>
  </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'nuxt-property-decorator';
import Network from '~/models/Network';

@Component({
  name: 'NetworkCard'
})
export default class Card extends Vue {
  @Prop({ type: Object })
  private network!: Network;

  get type(): String {
    return this.network.type === 'WIFI' ? 'wifi' : 'ethernet-cable';
  }
}
</script>

<style lang="scss" scoped>
.card-width {
  max-width: 250px;
  cursor: pointer;
}
</style>
