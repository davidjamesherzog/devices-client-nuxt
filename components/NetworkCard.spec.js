import { createLocalVue, shallowMount } from '@vue/test-utils';
import Buefy from 'buefy';
import NetworkCard from '~/components/NetworkCard.vue';

describe('NetworkCard', () => {
  describe('type', () => {
    test('should return wifi', () => {
      const networkCard = new NetworkCard();
      networkCard.network = {
        type: 'WIFI'
      };
      expect(networkCard.type).toEqual('wifi');
    });

    test('should return ethernet-cable', () => {
      const networkCard = new NetworkCard();
      networkCard.network = {
        type: 'ETHERNET'
      };
      expect(networkCard.type).toEqual('ethernet-cable');
    });
  });

  describe('template', () => {
    let localVue;

    beforeEach(() => {
      localVue = createLocalVue();
      localVue.use(Buefy);
    });

    test('should render network card', () => {
      const wrapper = shallowMount(NetworkCard, {
        localVue,
        propsData: {
          network: {
            name: 'TestWifi',
            description: 'Test description'
          }
        }
      });
      const title = wrapper.find('[data-card-title]');
      expect(title.text()).toEqual('TestWifi');
      const description = wrapper.find('[data-card-description]');
      expect(description.text()).toEqual('Test description');
      expect(wrapper).toMatchSnapshot();
    });
  });
});
