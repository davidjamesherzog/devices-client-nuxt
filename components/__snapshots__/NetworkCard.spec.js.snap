// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NetworkCard template should render home link 1`] = `
<div class="column card-width">
  <div class="card">
    <header class="card-header">
      <p data-card-title="" class="card-header-title has-text-grey">
        TestWifi
      </p>
    </header>
    <div class="card-content">
      <div class="content has-text-centered">
        <b-icon-stub type="is-primary" icon="ethernet-cable" size="is-large" data-card-icon=""></b-icon-stub>
      </div>
    </div>
    <footer class="card-footer">
      <div class="card-footer-item"><span data-card-description="">
          Test description
        </span></div>
    </footer>
  </div>
</div>
`;
